src/sentry/models/integrations/integration.py

```diff
@@ -85,10 +85,10 @@ def delete(self, *args, **kwds):
         with outbox_context(
             transaction.atomic(using=router.db_for_write(OrganizationIntegration)), flush=False
         ):
-            for organization_integration in self.organizationintegration_set.all():
-                organization_integration.delete()
             for outbox in Integration.outboxes_for_update(self.id):
                 outbox.save()
+            for organization_integration in self.organizationintegration_set.all():
+                organization_integration.delete()
             return super().delete(*args, **kwds)
```

Comment:
So this error was silent before, but what would happen is that we'd delete all the `OrgIntegrations` then attempt to use them to generate regions we should create outboxes directed at. That query would always return empty, but in monolith, we default to adding the `MONOLITH_REGION` so it didn't fail for prod or many tests. Now, we're generating the outbox before deleting the `OrgIntegration` so the issue shouldn't come up anymore.